### PR TITLE
Refactor external database handling

### PR DESCRIPTION
**Pull request by sutaakar**
Testing with external databases is completely rewritten under the hood (previous implementation was outdated and didn't use dedicated drivers for supported databases).

In case a custom driver is needed the scenario deploys own Docker registry into the project, downloads driver binary, builds Docker image with the driver and installs it into the registry, creates an image stream pointing to the installed image and use this image stream to build Kie server image with the driver.

Part of this PR is an update of XTF to 0.9 to support oc binary commands. Considering that XTF 0.9 is still a SNAPSHOT the Jenkins testing is limited until https://github.com/jboss-integration/bxms-qe-tests/pull/1548 is merged.

Feel free to take a look and please provide feedback.

### CODE REVIEW

File: framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/KieServerWithExternalDatabaseScenarioImpl.java [hunk 1]
```diff
@@ -52,6 +69,26 @@ public KieServerWithExternalDatabaseScenarioImpl(Map<String, String> envVariable
     @Override public void deploy() {
         super.deploy();
 
+        ExternalDatabase externalDatabase = ExternalDatabaseProvider.getExternalDatabase();
+        envVariables.putAll(externalDatabase.getExternalDatabaseEnvironmentVariables());
+
+        if (externalDatabase.getExternalDriver().isPresent()) {
```
Comment (jhrcek) [inline framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/KieServerWithExternalDatabaseScenarioImpl.java]: This pattern 

```java
if(optional.isPresent()) {
   val = optional.get()
   // do something with val
}
```

is better expressed with

```java
optional.ifPresent(val -> {
  //do something with val
}
```

File: framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/database/driver/Db2ExternalDriver.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public String getImageVersion() {
+        return "10.5";
+    }
+
+    @Override
+    public String getDockerFileRelativePath() {
+        return "db2-driver-image/Dockerfile";
```
Comment (jhrcek) [inline framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/database/driver/Db2ExternalDriver.java]: Nitpick: The "/Dockerfile" piece is repeated for each subclass of AbstractExternalDriver. Shouldn't it be extracted to the superclass, something like
```
File kieJdbcDriverDockerDir = new File(kieJdbcDriverScriptsFolder, getDockerFileRelativePath())
File kieJdbcDriverDockerFile = new File(kieJdbcDriverDockerDir, "Dockerfile");
```

**Comment by sutaakar** [comment]
@jhrcek Thanks for the tips, PR updated.

**Comment by sutaakar** [comment]
Made minor adjustments - reverted XTF version back to 0.6 (all required functionality is already there).
Tested on custom DB matrix job.
Ready to be merged.

**Comment by sutaakar** [comment]
retest this please

**Comment by sutaakar** [comment]
Test failures not related

**Pull request merged**